src/libteam/patch/0016-block-retry-count-changes.patch

```diff
@@ -0,0 +1,141 @@
+-			lacpdu_init(&lacpdu, 0xf1);
++			lacpdu_init(&lacpdu, lacp_port->last_received_lacpdu_version);
+ 		} else {
+ 			lacpdu_init(&lacpdu, 0x01);
+ 		}
+@@ -1483,9 +1486,12 @@ static int lacpdu_send(struct lacp_port *lacp_port)
+ 	return err;
+ }
+ 
++#define LACP_RETRY_COUNT_RESET_TIMEOUT_SECONDS 60
++
+ static int lacpdu_process(struct lacp_port *lacp_port, struct lacpdu* lacpdu)
+ {
+ 	int err;
++	struct timespec monotonic_time = {0};
+ 
+ 	if (!lacpdu_check(lacpdu)) {
+ 		teamd_log_warn("malformed LACP PDU came.");
+@@ -1523,17 +1529,55 @@ static int lacpdu_process(struct lacp_port *lacp_port, struct lacpdu* lacpdu)
+ 					lacp_port->partner_retry_count,
+ 					lacpdu->v2.actor_retry_count);
+ 			lacp_port->partner_retry_count = lacpdu->v2.actor_retry_count;
++			if (clock_gettime(CLOCK_MONOTONIC, &monotonic_time)) {
++				err = errno;
++				teamd_log_err("%s: unable to get current time: %s", lacp_port->tdport->ifname, strerror(err));
++				return -err;
++			}
++		}
++		if (lacp_port->partner_retry_count != LACP_CFG_DFLT_RETRY_COUNT) {
++			// Reset the change time every time a 0xf1 packet comes in
++			if (clock_gettime(CLOCK_MONOTONIC, &monotonic_time)) {
++				err = errno;
++				teamd_log_err("%s: unable to get current time: %s", lacp_port->tdport->ifname, strerror(err));
++				return -err;
++			}
++			lacp_port->lacp->next_retry_count_change_time = monotonic_time.tv_sec + LACP_RETRY_COUNT_RESET_TIMEOUT_SECONDS;
+ 		}
+ 	} else {
+ 		if (lacp_port->partner_retry_count != LACP_CFG_DFLT_RETRY_COUNT) {
+-			teamd_log_dbg(lacp_port->ctx, "%s: retry count from partner changed from %u to %u",
+-					lacp_port->tdport->ifname,
+-					lacp_port->partner_retry_count,
+-					LACP_CFG_DFLT_RETRY_COUNT);
+-			lacp_port->partner_retry_count = LACP_CFG_DFLT_RETRY_COUNT;
++			if (clock_gettime(CLOCK_MONOTONIC, &monotonic_time)) {
++				err = errno;
++				teamd_log_err("%s: unable to get current time: %s", lacp_port->tdport->ifname, strerror(err));
++				return -err;
++			}
++			if (monotonic_time.tv_sec < lacp_port->lacp->next_retry_count_change_time) {
++				teamd_log_dbg(lacp_port->ctx, "%s: ignoring resetting retry count to 3",
++						lacp_port->tdport->ifname);
++			} else {
++				teamd_log_dbg(lacp_port->ctx, "%s: retry count from partner changed from %u to %u",
++						lacp_port->tdport->ifname,
++						lacp_port->partner_retry_count,
++						LACP_CFG_DFLT_RETRY_COUNT);
++				lacp_port->partner_retry_count = LACP_CFG_DFLT_RETRY_COUNT;
++			}
+ 		}
+ 	}
+ 
++	if (lacp_port->last_received_lacpdu_version != lacpdu->version_number) {
++        teamd_log_dbg(lacp_port->ctx, "%s: LACPDU version changed from %u to %u",
++                lacp_port->tdport->ifname,
++                lacp_port->last_received_lacpdu_version,
++                lacpdu->version_number);
++		lacp_port->last_received_lacpdu_version = lacpdu->version_number;
++		// Force-send a LACPDU packet acknowledging change in version
++		err = lacpdu_send(lacp_port);
```

Comment:
The same packet w/ changed version is sent here. Don't you also want to change retry count in the sent packet?

Reply:
The new retry count is in this packet (it would've been updated in the previous if-else block, the `lacp_port->partner_retry_count` field).